daemon/policy.go

```diff
@@ -123,8 +120,6 @@ func (d *Daemon) UpdatePolicyEnforcement(e *endpoint.Endpoint) bool {
 			endpointLabels = append(endpointLabels, lbl)
 		}
 		e.Mutex.RUnlock()
-		d.GetPolicyRepository().Mutex.RLock()
-		defer d.GetPolicyRepository().Mutex.RUnlock()
```

Comment:
Discussed over Slack - UpdatePolicyEnforcement is not holding the lock, but GetRulesMatching is.